### PR TITLE
TSK-146: Introduced ClassificationSummary Interface and Impl.

### PR DESCRIPTION
**Pull request by MLengl**
Also deleted forbidden methods on Classification-Interface. 
Mappers for Query changed.

### CODE REVIEW

File: lib/taskana-core/src/main/java/pro/taskana/ClassificationQuery.java [hunk 1]
```diff
@@ -41,36 +45,44 @@
 
     /**
      * Add your domain to your query.
-     * @param domain TODO
+     *
+     * @param domain
+     *            TODO
```
Comment (BerndBreier) [inline lib/taskana-core/src/main/java/pro/taskana/ClassificationQuery.java]: Please fill in javadoc 'TODO' if modifying a class / interface

File: lib/taskana-core/src/main/java/pro/taskana/ClassificationQuery.java [hunk 2]
```diff
@@ -41,36 +45,44 @@
...
      * @return TODO
      */
     ClassificationQuery domain(String... domain);
 
     /**
      * Add to your query if the Classification shall be valid in its domain.
-     * @param validInDomain TODO
+     *
+     * @param validInDomain
+     *            TODO
```
Comment (BerndBreier) [inline lib/taskana-core/src/main/java/pro/taskana/ClassificationQuery.java]: see above

File: lib/taskana-core/src/main/java/pro/taskana/ClassificationQuery.java [hunk 3]
```diff
@@ -41,36 +45,44 @@
...
      * @return TODO
      */
     ClassificationQuery validInDomain(Boolean validInDomain);
 
     /**
      * Add your created-Dates to your query.
-     * @param created TODO
+     *
+     * @param created
+     *            TODO
```
Comment (BerndBreier) [inline lib/taskana-core/src/main/java/pro/taskana/ClassificationQuery.java]: also here

File: lib/taskana-core/src/main/java/pro/taskana/ClassificationSummary.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package pro.taskana;
```
Comment (BerndBreier) [inline lib/taskana-core/src/main/java/pro/taskana/ClassificationSummary.java]: Is it really required to have interface + implementation for summaries?
TaskSummary and WorkbasketSummary are simple Java Beans in the pro.taskana.model folder.
What is the reason for I/F + Impl for ClassificationSummary?

Comment (MLengl) [reply lib/taskana-core/src/main/java/pro/taskana/ClassificationSummary.java]: All model-classes we´ve got implements an interface for the user-api. So all summaries should implement an interface, too - i´ve asked Holger before i´ve done it this way.

**Comment by MLengl** [comment]
Commit is updated and complete.

**Pull request merged**